pkg/spire/spire_test.go

```diff
@@ -625,79 +618,47 @@ func TestSpire_TaskRunResultsSignTamper(t *testing.T) {
 				}
 				return prs
 			},
-			success: false,
+			wantErr: true,
 		},
 	}
 
 	for _, tt := range testCases {
-		ctx := context.Background()
-		for _, tr := range testTaskRuns() {
-			results := genPr()
-			success := func() bool {
-				resp := &fakeworkloadapi.X509SVIDResponse{
+		t.Run(tt.desc, func(t *testing.T) {
+			ctx := context.Background()
+			for _, tr := range testTaskRuns() {
+				results := pr
+
+				wl.SetX509SVIDResponse(&fakeworkloadapi.X509SVIDResponse{
 					Bundle: ca.X509Bundle(),
-					SVIDs:  makeX509SVIDs(ca, spiffeid.RequireFromPath(td, getTaskrunPath(tr))),
-				}
-				wl.SetX509SVIDResponse(resp)
+					SVIDs:  x509svids(ca, spiffeid.RequireFromPath(td, taskrunPath(tr))),
+				})
 
-				sigResults, err := ec.Sign(ctx, results)
-				if err != nil {
-					return false
-				}
+				var err error
+				var sigResults []v1beta1.PipelineResourceResult
+				if sigResults, err = ec.Sign(ctx, results); err == nil {
+					results = append(results, sigResults...)
+					if tt.tamperFn != nil {
+						results = tt.tamperFn(results)
```

Comment:
...in fact it looks like this test *does* end up modifying `results` in place, because many of the `tamperFn`s do! This will only translate into a modification of `pr` if `sigResults` is empty, because a slice literal's capacity is guaranteed to equal its length, and this might not be compatible with `err == nil`, but at that point we're relying on the correctness of the code *under* test to show the correctness of the test *itself*, which is a problem. (Alternatively, one could argue that the length of this chain of reasoning is already a problem.)

Reply:
this is resolved by properly in-lining the definition of the `results` local var